### DISCUSSION TITLE
drawChar() function with loaded font

### DISCUSSION THREAD

**Discussion by laci222** [Bodmer/TFT_eSPI#1197] [Q&A - Fonts]
Hi Bodmer and other experts, 

I want to print non-English alphabet characters like Á, É, Ű to the screen. I don't have them in a String variable, but in unicode codepoint format in uint16_t (like uint16_t = 0x00C1 is Á). In reality, I have an array of these characters. Do you know any way how can I convert them to either String or a char array (according to this site it is possible to store these characters in a char array: https://playground.arduino.cc/Code/UTF-8/  char good[] = "5µA";   // Good, and also this library has a Unicode_test example in string format) Also the print functions work with codepoint formats like this: tft.println("\u00C1"); // Á but I don't know if I can convert a uint16_t to this format. 

As I had no luck converting them, I found the drawChar(uint16_t uniCode, int32_t x, int32_t y) function. This function can draw a Unicode glyph onto the screen. This accepts my characters and tries to print them, but the default font set does not contain the characters, so they are not shown properly. **I converted a font to .vlw format and the to binary and loaded them, but this function still tries to use the original default font.** 

This is the part of the code, where I try to use the special font: 

    tft.loadFont(ArielMT_12);
    tft.setTextSize(3);
    String strFormat= "Hell_áÁ$/^do Bello";
    tft.print(strFormat);
    for (size_t i = 0; i < length; i++) {
      tft.drawChar(text[i], i*20, 40);
    }
    tft.unloadFont();

The strFormat string is displayed properly, as it uses the loaded ArielMT_12 font and the special characters are great too. 
But the characters that were displayed with the drawChar() function uses the default font (with TextSize 3) and the special characters are not displayed, only the ASCII ones. 

Can you please help, how to make this function use the loaded font set, or how to convert the codepoint format to String or to basic char array?

**Comment by Bodmer** [comment]
drawChar is not supported for smooth fonts.
Use drawString or print.
Another option is to use setCursor followed by drawGlyph.

**Reply by laci222** [reply]
Thanks for the tip, drawGlyph is working properly :) 
Is there a way in your library to convert an array of codepoints (uint16_t array) to string or char array? 
Is any of the built-in fonts support non-english characters, or if I want to print É, Á, Ű, I have to add my own font?
